### DISCUSSION TITLE
Does edsl use the `n` argument when generating multiple responses to identical question-persona pairs with OpenAI?

### DISCUSSION THREAD

**Discussion by MaxGhenis** [expectedparrot/edsl#1123] [Q&A]
The OpenAI API supports a `n` argument which generates multiple (varied) results for the same query. This is faster and cheaper than separate queries, including async, because it only sends the input tokens once.

I tried looking through the code to see if edsl does this when users request multiple responses per question-persona pair, but couldn't tell. If not, could be worth adding. It would add complexity since I believe it's only available in OpenAI, not Anthropic or Gemini.

**Comment by rbyh** [comment]
Thanks for raising this!

You may have not noticed this functionality for generating multiple results for a survey when you run it (I see that it does not come up for any relevant search terms at the docs page, unfortunately--I will try to fix that!):

https://docs.expectedparrot.com/en/latest/results.html#generating-multiple-results

However, this is not doing what you are suggesting. We will look into it!
